test/gateway/eip20_gateway/constructor.js

```diff
@@ -76,12 +76,21 @@ contract('EIP20Gateway.constructor() ', (accounts) => {
       burner,
     );
 
+    const valueTokenAddress = await gateway.valueToken.call();
+
+    assert.equal(
+      valueTokenAddress,
+      mockToken.address,
+      'Invalid valueTokenAddress address from contract.',
+    );
+
+    // token supports previous ABIs
```

Comment:
Should probably be a separate test with its own `it`.

Reply:
@schemar: I don't think we typically have separate tests for getters. Or is your concern that this does not fall under the given test caption, "should pass with correct parameters"? Or is your concern because this function exists for a special reason?

Setting aside the caption, I think the passing constructor test is the appropriate location for testing a getter with an invariant value, because construction is the only call through which the value changes.

Thoughts?

Reply:
My concern was that "backwards compatibility" does not fall under "correct parameters". That's why I suggest a different test about backwards compatibility.

Reply:
Agreed! But I would argue the test caption should be different. E.g., [in a BrandedToken test](https://github.com/OpenST/brandedtoken-contracts/blob/develop/test/branded_token/constructor.js#L103): `it('Successfully sets state variables'`

If the test caption is the only significant reason, then I'd prefer to leave things in their current locations.

Reply:
Yes. If the test is only about testing the variable assignment in the constructor, then I agree 👍